### ISSUE TITLE
superTokensInstance should not be package-level

### ISSUE THREAD

**Issue by Leskodamus** [supertokens/supertokens-golang#420]
The current implementation defines and uses a package-level variable for the `*superTokens` instance
which is not a great if anyone would like to have multiple instances or for testing (in parallel). 

Instead it should be handled in encapsulation, for example by changing the `func supertokensInit(config TypeInput) error {}` and `func Init(config TypeInput) error {}` functions to return a `*superTokens` object and by making `type superTokens struct {}` a public struct:

```Go
type SuperTokens struct {}

func supertokensInit(config TypeInput) (error, *SuperTokens) {}

func Init(config TypeInput) (error, *SuperTokens) {}
```
Maybe `Init(config TypeInput)` should even be renamed to `NewSuperTokens` or `NewSuperTokensInstnace`.

To me this implementation is not very idiomatic Go and looks more like JavaScript.

**Comment by rishabhpoddar** [comment]
Agreed. Until this is done though, you can call the `supertokens.supertokens.ResetForTest()` function during tests so that you can call init multiple times.
